src/cascade/model/grids.py

```diff
@@ -0,0 +1,235 @@
+"""Support for dismod concepts that are expressed over grids: age-time, smoothings and value priors.
+"""
+from math import isclose, isinf
+
+import numpy as np
+
+GRID_SNAP_DISTANCE = 1 / 365
+
+
+def _nearest(values, query):
+    values = np.array(values)
+    return values[np.abs(values - query).argmin()]
+
+
+class AgeTimeGrid:
+    """Structure for storing age and times over which the model will estimate.
+    Also constructors for making standard age-time grids.
+    """
+
+    @classmethod
+    def uniform(cls, age_start, age_end, age_step, time_start, time_end, time_step):
+        """Construct an age-time grid with uniform spacing on each axis.
+        All values are in units of years.
+
+        Args:
+            age_start: the lowest age in the grid
+            age_end: the highest age in the grid
+            age_step: the spacing of age points
+            time_start: the earliest time in the grid
+            time_end: the latest time in the grid
+            time_step: the spacing of time points
+        """
+        ages = np.arange(age_start, age_end, age_step)
```

Comment:
I have a slight preference for linspace here. Let me sell you on that: Both linspace and arange require that the user be careful about matching step sizes to beginning and end. With linspace, you are guaranteed that the end is included in the range, but with arange you are guaranteed that the step sizes are right. I think it's more important to go to age 100 than to have exactly 2-year intervals along the way. Hmmm. I don't know. both are annoying.

Reply:
I'm fine with either. This is a tool I made up based on my assumptions of what we might want. It looks like in practice this won't be very useful as the modelers will usually be explicitly setting up their grids. So I might argue for pulling this method completely.

Reply:
I suspect we will have things that generate age ranges but that they will be more complicated because the young ages will have finer grid points and the oldest more granular regions. Removing this from the class is kinda fine, I suppose.

Reply:
I'm going to keep it and switch to linspace. Even if the modelers don't use it, it's nice for examples.

Reply:
Ok. After realizing the difficulty of having non-integer/simple float points in the grids I'm going back to arange.